docs/ops/movement/Gather_7.md

```diff
@@ -0,0 +1,133 @@
+
+Where `data`, `indices` and `axis` are tensors from first, second and third inputs correspondingly, and `b` is 
+the number of batch dimensions.
+
+**Attributes**:
+* *batch_dims*
+  * **Description**: *batch_dims* (also denoted as `b`) is a leading number of dimensions of `data` tensor and `indices` 
+  representing the batches, and *Gather* starts to gather from the `b` dimension. It requires the first `b` 
+  dimensions in `data` and `indices` tensors to be equal.
+  * **Range of values**: `[0; min(data.rank, indices.rank))` and `batch_dims <= axis`
+  * **Type**: *T_IND*
+  * **Default value**: 0
+  * **Required**: *no*
+
+Example 1 with default *batch_dims* value:
+```
+batch_dims = 0
+axis = 0
+
+indices = [0, 0, 4] 
+data    = [1, 2, 3, 4, 5]
+output  = [1, 1, 5]
+```
+
+Example 2 with non-default *batch_dims* value:
+```
+batch_dims = 1
+axis = 1
+
+indices = [[0, 0, 4], <-- this is applied to the first batch 
+           [4, 0, 0]]  <-- this is applied to the second batch
+indices_shape = (2, 3)
+
+data    = [[1,  2,  3,  4, 5],  <-- the first batch
+           [6,  7,  8,  9, 10]]  <-- the second batch 
+data_shape = (2, 5)
+
+output  = [[ 1,  1,  5],
+           [10,  6,  6]]
+output_shape = (2, 3)
+```
+
+Example 3 with non-default *batch_dims* value:
+```
+batch_dims = 2
+axis = 2
+
+indices = [[[0, 0, 4],  <-- this is applied to the first batch, index = (0, 0)
+            [4, 0, 0]],  <-- this is applied to the second batch, index = (0, 1)
+          
+           [[1, 2, 4],  <-- this is applied to the third batch, index = (1, 0)
+            [4, 3, 2]]]  <-- this is applied to the fourth batch, index = (1, 1) 
+indices_shape = (2, 2, 3)
+
+data    = [[[1,  2,  3,  4,  5],  <-- the first batch, index = (0, 0)
+            [6,  7,  8,  9, 10]],  <-- the second batch, index = (0, 1)
+          
+           [[11,  12,  13,  14, 15],  <-- the third batch, index = (1, 0)
+            [16,  17,  18,  19, 20]]]  <-- the fourth batch, index = (1, 1)
+data_shape = (2, 2, 5)
+
+output  = [[[ 1,  1,  5],
+            [10,  6,  6]],
+
+           [[12, 13, 15],
+            [20, 19, 18]]] 
+output_shape = (2, 2, 3)
+```
+
+**Inputs**
+
+* **1**:  `data` tensor of type *T* with arbitrary data. **Required**.
+
+* **2**:  `indices` tensor of type *T_IND* with indices to gather. The values for indices are in the range `[0, data[axis] - 1]`. 
+**Required**.
+
+* **3**:  Scalar or 1D tensor `axis` of *T_IND* type is a dimension index to gather data from. For example, 
+*axis* equal to 1 means that gathering is performed over the first dimension. Negative value means reverse indexing. 
+Allowed values are from `[-len(data.shape), len(indices.shape) - 1]` and `axis >= batch_dims`. 
+**Required**.
+
+**Outputs**
+
+* **1**: The resulting tensor of type *T* that consists of elements from `data` tensor gathered by `indices`. Shape of 
+the output tensor is `data.shape[:axis] + indices.shape[batch_dims:] + data.shape[axis + 1:]`
```

Comment:
the output shape seems incorrect here because the first dimensions must be `indices.shape[:batch_dims]`

Reply:
This formula is right. You are also right that first dimensions are `indices.shape[:batch_dims]` but since `axis >= batch_dims` and batch dimensions are of the same shape then again goes `data.shape[batch_dims:axis] +...` so it's better to write at once in a single expression than:
`indices.shape[:batch_dims] + data.shape[batch_dims:axis] + indices.shape[batch_dims:] + data.shape[axis + 1:]`